From fd5d9d005973034c444bc018ebf69cd01bc9cdab Mon Sep 17 00:00:00 2001
From: Amitesh Anand <quic_amitesh@quicinc.com>
Date: Wed, 13 Mar 2024 20:27:34 +0530
Subject: [PATCH] net: Fix usage of uninitialized object for flushing conntrack

This fixes patch from Openwrt community:

OpenWrt: 600-netfilter_conntrack_flush.patch
netfilter: add support for flushing conntrack via /proc
lede-commit 8193bbe59a74d34d6a26d4a8cb857b1952905314

We must initialize nf_ct_iter_data object before calling
nf_ct_iterate_cleanup_net(); else kernel receives garbage value
for port_id and report

Change-Id: Iff7ff59ca8d7c59e3def9396832fa9ff72698cb5
Signed-off-by: Amitesh Anand <quic_amitesh@quicinc.com>
---
 net/netfilter/nf_conntrack_standalone.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/netfilter/nf_conntrack_standalone.c b/net/netfilter/nf_conntrack_standalone.c
index d46fb718d1f4..256446d359a9 100644
--- a/net/netfilter/nf_conntrack_standalone.c
+++ b/net/netfilter/nf_conntrack_standalone.c
@@ -492,7 +492,7 @@ static int kill_matching(struct nf_conn *i, void *data)
 static int ct_file_write(struct file *file, char *buf, size_t count)
 {
 	struct seq_file *seq = file->private_data;
-	struct nf_ct_iter_data iter_data;
+	struct nf_ct_iter_data iter_data = { };
 	struct kill_request kr = { };
 
 	if (count == 0)
-- 
2.34.1

